clients/privacy-center/packages/fides-consent/src/lib/consent-value.ts

```diff
@@ -0,0 +1,43 @@
+import { ConsentContext } from "./consent-context";
+
+export type ConditionalValue = {
+  value: boolean;
+  globalPrivacyControl: boolean;
+};
+
+/**
+ * A consent value can be a boolean:
+ *  - `true`: consent/opt-in
+ *  - `false`: revoke/opt-out
+ *
+ * A consent value can also be context-dependent, which means it will be decided based on
+ * information about the user's environment (browser). The `ConditionalValue` object maps the
+ * context conditions to the value that should be used:
+ *  - `value`: The default value if no context applies.
+ *  - `globalPrivacyControl`: The value to use if the user's browser has Global Privacy Control
+ *    enabled.
+ */
+export type ConsentValue = boolean | ConditionalValue;
+
+export type ConsentTypeToValue = {
+  [consentType: string]: ConsentValue;
+};
+
+export const resolveConsentValue = (
+  value: ConsentValue | undefined,
+  context: ConsentContext
+): boolean => {
+  if (value === undefined) {
+    return false;
+  }
+
+  if (typeof value === "boolean") {
```

Comment:
Is it possible for `value` to be a boolean and also `context.globalPrivacyControl === true`? I'm wondering if you set this condition to be read first purposefully. Reading the comment above, it seems like it's not something we need to account for, but curious.

Reply:
If `value` is not `undefined`, then the user has made an explicit choice, which takes precedence over the "default" choice based on the browser context. https://github.com/ethyca/fides/issues/2230 will cover the UI indication that their explicit choice conflicts with the expected default.